kimchi/src/circuits/polynomials/foreign_field_add/circuitgates.rs

```diff
@@ -0,0 +1,233 @@
+//~     bound_carry_lo  -> k0
+//~     bound_carry_mi  -> k1
+//~```
+//~
+//~ Let `left_input_lo`, `left_input_mi`, `left_input_hi` be 88-bit limbs of the left element
+//~
+//~ Let `right_input_lo`, `right_input_mi`, `right_input_hi` be 88-bit limbs of the right element
+//~
+//~ Let `foreign_modulus_lo`, `foreign_modulus_mi`, `foreign_modulus_hi` be 88-bit limbs of the foreign modulus
+//~
+//~ Then the limbs of the result are
+//~
+//~ - `result_lo = left_input_lo +/- right_input_lo - field_overflow * foreign_modulus_lo - 2^{88} * result_carry_lo`
+//~ - `result_mi = left_input_mi +/- right_input_mi - field_overflow * foreign_modulus_mi - 2^{88} * result_carry_mi + result_carry_lo`
+//~ - `result_hi = left_input_hi +/- right_input_hi - field_overflow * foreign_modulus_hi + result_carry_mi`
+//~
+//~ `field_overflow` $=0$ or $1$ or $-1$ handles overflows in the field
+//~
+//~ `result_carry_i` $= -1, 0, 1$ are auxiliary variables that handle carries between limbs
+//~
+//~ Apart from the range checks of the chained inputs, we need to do an additional range check for a final bound
+//~ to make sure that the result is less than the modulus, by adding `2^{3*88} - foreign_modulus` to it.
+//~ (This can be computed easily from the limbs of the modulus)
+//~ Note that `2^{264}` as limbs represents: (0, 0, 0, 1) then:
+//~
+//~ The upper-bound check can be calculated as
+//~ - `bound_lo = result_lo - foreign_modulus_lo - bound_carry_lo * 2^{88}`
+//~ - `bound_mi = result_mi - foreign_modulus_mi - bound_carry_mi * 2^{88} + bound_carry_lo`
+//~ - `bound_hi = result_hi - foreign_modulus_hi + 2^{88} + bound_carry_mi`
+//~
+//~ `bound_carry_i` $= 0$ or $1$ or $-1$ are auxiliary variables that handle carries between limbs
+//~
+//~ The range check of `bound` can be skipped until the end of the operations
+//~ and `result` is an intermediate value that is unused elsewhere (since the final `result`
+//~ must have had the right number of moduli subtracted along the way).
+//~
+//~ You could lay this out as a double-width gate for chained foreign additions and a final row, e.g.
+//~
+//~ | col | `ForeignFieldAdd`       | more `ForeignFieldAdd` | or `ForeignFieldFin` |
+//~ | --- | ----------------------- | ---------------------- | -------------------- |
+//~ |   0 | `left_input_lo`  (copy) | `result_lo` (copy)     | `resmin_lo` (copy)   |
+//~ |   1 | `left_input_mi`  (copy) | `result_mi` (copy)     | `resmin_mi` (copy)   |
+//~ |   2 | `left_input_hi`  (copy) | `result_hi` (copy)     | `resmin_hi` (copy)   |
+//~ |   3 | `right_input_lo` (copy) |  ...                   | `bound_lo`  (copy)   |
+//~ |   4 | `right_input_mi` (copy) |  ...                   | `bound_mi`  (copy)   |
+//~ |   5 | `right_input_hi` (copy) |  ...                   | `bound_hi`  (copy)   |
+//~ |   6 | `field_overflow`        |  ...                   |  -                   |
+//~ |   7 | `carry_lo`              |  ...                   | `bound_carry_lo`     |
```

Comment:
I think you can call these just `carry_*` since we know it's the bound and that also matches the variable names in `ForeignFieldFin` implementation.

Reply:
ok as you wish, I was just trying to be "different" so that the ones in the FFadd row are not confused with the ones of the FFfin row.

Reply:
then I guess I can switch the names inside ForeignFieldFin implementation instead.